CSRF vulnerability fix

atai opened the pull request:
<!-- Please describe your pull request here. -->

### Testing done

- CSRF vulnerability and missing permission check — Fixed in doOwnersSubmit and doProjectSpecificSecuritySubmit methods https://www.jenkins.io/security/advisory/2022-03-29/#SECURITY-2062%20(1)
- CSRF vulnerability — Fixed in doRestoreDefaultSpecificSecuritySubmit method https://www.jenkins.io/security/advisory/2022-03-29/#SECURITY-2062%20(2)
- Update Jenkins version — Updated from Jenkins 1.x to 2.377
- Use plugin bill of materials — Implemented BOM (bom-2.361.x) for dependency management
- Added security tests to verify CSRF protection and permission checks
- Updated deprecated APIs and patterns
- Improved code quality and test coverage

<!-- Comment:
Provide a clear description of how this change was tested.
At minimum this should include proof that a computer has executed the changed lines.
Ideally this should include an automated test or an explanation as to why this change has no tests.
Note that automated test coverage is less than complete, so a successful PR build does not necessarily imply that a computer has executed the changed lines.
If automated test coverage does not exist for the lines you are changing, you must describe the scenario(s) in which you manually tested the change.
For frontend changes, include screenshots of the relevant page(s) before and after the change.
For refactoring and code cleanup changes, exercise the code before and after the change and verify the behavior remains the same.
-->

### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests that demonstrate the feature works or the issue is fixed

<!--
Put an `x` into the [ ] to show you have filled the information.
The template comes from https://github.com/jenkinsci/.github/blob/master/.github/pull_request_template.md 
You can override it by creating .github/pull_request_template.md in your own repository 
-->

MarkEWaite commented:
Please update the Jenkinsfile when you update the Jenkins version so that ci.jenkins.io will test with the same version as is used in the code.  The "Improve a plugin" tutorial includes a step for the Jenkinsfile.

MarkEWaite commented on pom.xml:
```diff
@@ -23,10 +24,10 @@
-        <jenkins.version>1.651.3</jenkins.version>
-        <java.level>7</java.level>
+        <jenkins.version>2.361.4</jenkins.version>
+        <java.level>17</java.level>
```
The improve a plugin tutorial advises that this should be removed, since it is no longer relevant.
```suggestion
```

MarkEWaite commented on pom.xml:
```diff
@@ -23,10 +24,10 @@
     <properties>
-        <jenkins.version>1.651.3</jenkins.version>
-        <java.level>7</java.level>
+        <jenkins.version>2.361.4</jenkins.version>
```
This choice of Jenkins version goes against the guidance in ["Choosing a Jenkins version to build against"](https://www.jenkins.io/doc/developer/plugin-development/choosing-jenkins-baseline/).  Jenkins 2.479.3 is a better choice because it requires Java 17 and is much closer to recent Jenkins versions.  Jenkins 2.504.3 is also a good choice.
```suggestion
        <jenkins.baseline>2.479</jenkins.baseline>
        <jenkins.version>${jenkins..version}.3</jenkins.version>
```

mawinter69 replied:
I second that. Consider that 2.361.4 is over 3 years old.

MarkEWaite commented on pom.xml:
```diff
@@ -42,12 +43,24 @@
+            <dependency>
+                <groupId>io.jenkins.tools.bom</groupId>
+                <artifactId>bom-2.361.x</artifactId>
+                <version>2025.v816d28f1e04f</version>
```
If you are really intending to continue testing and supporting Jenkins versions that supported Java 11, then you should choose the most recent release of the 2.361.x plugin BOM, as noted in the ["depending on older versions"](https://github.com/jenkinsci/bom?tab=readme-ov-file#depending-on-older-versions) documentation.

I think that 2.479.3 is a much better choice, since the Jenkins project has stopped supporting Java 11 a year or more ago.
```suggestion
                <version>2102.v854b_fec19c92</version>
```

atai replied:
I absolutely agree with your comments, and I'll address them. I should mention that upgrading from Jenkins 1.x was challenging given the 7+ year gap since the last update. I used AI tools to help navigate the large codebase, but as you correctly noted, the volume of changes is substantial, and I understand it's not ideal to address everything at once.

atai commented:
Dear Mark,
Thank you for the comments. I've noted them and will address them.

Just FYI: the Jenkinsfile upgrade has been in the queue since 2024 (https://github.com/jenkinsci/ownership-plugin/pull/90), which highlights the main issue: this plugin has no active maintainers.

I understand the PR is large, and I'm asking for maintenance access because:

- The plugin needs ongoing maintenance
- I may not be the ideal candidate, but I'm willing to step up
- I need these fixes for my servers

What I've done so far:

- Migrated from Jenkins 1.x to 2.377
- Fixed all 120+ tests
- Fixed critical security vulnerabilities (CSRF)
- Fixed UI issues (old images breaking page layout)
- Fixed SpotBugs errors you identified

My commitment:
If I can be confident the code will be merged and I can maintain the plugin going forward, allowing me to use plugin updates from Jenkins update stream, I'm ready to:

- Address all your comments
- Continue modernization work
- Act as maintainer (or at least ensure PRs don't sit for a year)

If the plugin remains unmaintained, I'll use a local HPI file, though that's not ideal. I'd prefer to contribute properly to the Jenkins ecosystem.

Thanks for your consideration. I'd appreciate if we could finalize the maintainer access request process. The request is tracked in https://github.com/jenkins-infra/repository-permissions-updater/pull/4864

MarkEWaite commented:
> My commitment:
> If I can be confident the code will be merged and I can maintain the plugin going forward, allowing me to use plugin updates from Jenkins update stream, I'm ready to:

That is good enough for me.  Thanks for your willingness to update the plugin and maintain it!

atai commented on src/main/resources/com/synopsys/arc/jenkins/plugins/ownership/jobs/JobOwnerJobAction/configure-project-specifics.jelly:
```diff
@@ -28,7 +28,7 @@
         <l:main-panel>
             <h1>
-                <img src="${imagesURL}/48x48/fingerprint.gif"/>
+                <l:icon class="icon-fingerprint icon-xlg" />
```
Yep - this doesn't work with this version. For the future improvements then.

oleg-nenashev commented on architectural_review_plan.plan.md:
```diff
@@ -0,0 +1,236 @@
+---
+name: Architectural Review Plan
+overview: Comprehensive architectural review of the Jenkins Ownership Plugin covering Jenkins best practices compliance, CRUD principles, and DRY violations with specific recommendations for improvements.
```
Is it actually needed?

atai replied:
From my point of view it is good to have for further steps. Moved to Issues.

oleg-nenashev commented:
Probably I should not have merged other accumulated PRs, but the merge conflict is quite mild.

@atai I am perfectly fine with you taking over the plugin if you're interested. And thanks for the contribution and long overdue patches!

oleg-nenashev commented:
> Remove Architectural Review Plan document as it is no longer relevant to the project. This file contained detailed compliance and improvement recommendations for the Jenkins Ownership Plugin.

For what it's worth, the content is still relevant. I'm just not sure having it within the repository is the right approach unless it supports AI workflows. Might be better to just have it as a number of GitHub issues.

atai commented:
> From MPOV you shall add more automated tests.

I agree. I've opened an issue https://github.com/jenkinsci/ownership-plugin/issues/97 

This PR is already quite large with the changes, so I'd prefer to proceed with tests in a separate PR.
